### ISSUE TITLE
Weird prediction results on custom 24 kps car dataset

### ISSUE THREAD

**Issue by KleinXin** [DuncanZauss/Keypoint_Communities#6]
I have a dataset with 91 trannning & 9 validation images where 24 car kps are annotated and annotations are transformed to coco format as apollo where keypoints array is of size 24*3. 

I trained the shufflenetv2k16 24kps model. Trainning seems correct as shown below 

```
INFO:openpifpaf.network.trainer:{'type': 'train', 'epoch': 1, 'batch': 0, 'n_batches': 18, 'time': 0.471, 'data_time': 4.657, 'lr': 2e-05, 'loss': 3540.623, 'head_losses': [3.702, 310.73, 0.799, 205.145, 3015.488, 4.761]}
INFO:openpifpaf.network.trainer:{'type': 'train', 'epoch': 1, 'batch': 11, 'n_batches': 18, 'time': 0.434, 'data_time': 0.0, 'lr': 2e-05, 'loss': 3779.676, 'head_losses': [8.102, 372.573, 0.679, 149.955, 3243.949, 4.419]}
INFO:openpifpaf.network.trainer:applying ema
INFO:openpifpaf.network.trainer:{'type': 'train-epoch', 'epoch': 2, 'loss': 3874.88336, 'head_losses': [2.99886, 334.91886, 0.84728, 231.23997, 3298.75681, 6.1216], 'time': 14.2, 'n_clipped_grad': 0, 'max_norm': 0.0}
INFO:openpifpaf.network.trainer:restoring params from before ema

...

INFO:openpifpaf.network.trainer:{'type': 'train', 'epoch': 100, 'batch': 0, 'n_batches': 18, 'time': 0.625, 'data_time': 2.206, 'lr': 2e-05, 'loss': 2648.88, 'head_losses': [-40.191, 469.212, 0.059, 123.052, 2095.671, 1.078]}
INFO:openpifpaf.network.trainer:{'type': 'train', 'epoch': 100, 'batch': 11, 'n_batches': 18, 'time': 0.427, 'data_time': 0.0, 'lr': 2e-05, 'loss': 2616.817, 'head_losses': [-36.641, 429.025, 0.159, 120.99, 2102.774, 0.509]}
INFO:openpifpaf.network.trainer:applying ema
INFO:openpifpaf.network.trainer:{'type': 'train-epoch', 'epoch': 101, 'loss': 3276.90862, 'head_losses': [-45.2643, 351.73443, 0.60137, 196.80784, 2769.34305, 3.68621], 'time': 14.2, 'n_clipped_grad': 0, 'max_norm': 0.0}
INFO:openpifpaf.network.trainer:restoring params from before ema

...

INFO:openpifpaf.network.trainer:{'type': 'train', 'epoch': 199, 'batch': 0, 'n_batches': 18, 'time': 0.46, 'data_time': 4.563, 'lr': 2e-06, 'loss': 3563.018, 'head_losses': [-61.656, 281.741, 0.614, 203.755, 3135.36, 3.203]}
INFO:openpifpaf.network.trainer:{'type': 'train', 'epoch': 199, 'batch': 11, 'n_batches': 18, 'time': 0.435, 'data_time': 0.0, 'lr': 2e-06, 'loss': 3455.956, 'head_losses': [-54.041, 327.122, 0.471, 192.066, 2987.241, 3.097]}
INFO:openpifpaf.network.trainer:applying ema
INFO:openpifpaf.network.trainer:{'type': 'train-epoch', 'epoch': 200, 'loss': 3159.92377, 'head_losses': [-52.87015, 321.86455, 0.44325, 169.9907, 2718.06543, 2.42999], 'time': 20.8, 'n_clipped_grad': 0, 'max_norm': 0.0}
INFO:openpifpaf.network.trainer:model written: outputs/shufflenetv2k16-211021-024942-apollo.pkl.epoch200

...

INFO:openpifpaf.network.trainer:{'type': 'train', 'epoch': 299, 'batch': 0, 'n_batches': 18, 'time': 0.614, 'data_time': 6.553, 'lr': 2e-07, 'loss': 3374.045, 'head_losses': [-54.347, 392.751, 0.522, 170.645, 2860.723, 3.751]}
INFO:openpifpaf.network.trainer:{'type': 'train', 'epoch': 299, 'batch': 11, 'n_batches': 18, 'time': 0.443, 'data_time': 0.0, 'lr': 2e-07, 'loss': 2769.724, 'head_losses': [-43.393, 481.309, 0.127, 131.402, 2198.701, 1.578]}
INFO:openpifpaf.network.trainer:applying ema
INFO:openpifpaf.network.trainer:{'type': 'train-epoch', 'epoch': 300, 'loss': 3054.62056, 'head_losses': [-52.68392, 345.90691, 0.38009, 155.51899, 2603.45154, 2.04698], 'time': 17.1, 'n_clipped_grad': 0, 'max_norm': 0.0}
INFO:openpifpaf.network.trainer:model written: outputs/shufflenetv2k16-211021-024942-apollo.pkl.epoch300
```

However, all predictions are empty array.

![image](https://user-images.githubusercontent.com/13084622/138217441-611923cf-953e-40bb-ade5-bb490b89179a.png)

The prediction codes are shown below

```
import os
import numpy as np
from PIL import Image
import cv2
import openpifpaf

if __name__ == "__main__":

   src_img_folder = './images/train'
   dst_img_folder = './res/train'
   weights = './outputs/shufflenetv2k16-211021-024942-apollo.pkl.epoch200' 
 
   predictor = openpifpaf.Predictor(checkpoint=weights)

   img_names = os.listdir(src_img_folder)
   img_names = sorted(img_names)

   for name_idx,name_ in enumerate(img_names):

       I = Image.open(os.path.join(src_img_folder,name_)).convert('RGB')

       predictions, gt_anns, image_meta = predictor.pil_image(I)

       print(f"========== predictions: {predictions} ===============")
       print(f"========== image_meta: {image_meta} ==============")
```

**Comment by KleinXin** [comment]
@DuncanZauss 

My 24 car kps have the same definition as yours. The predictions on trainning set are also empty which means the model did not fit the dataset very well. I think it should be overfitting with such a small dataset. The trainning & validation sets are randomly splitted from 100 images.

Now I have 500 images and will try it again. Feel free to figure out the abnormalities in both the trainning log and inference codes. Thx!
